source/create.js

```diff
@@ -51,18 +53,19 @@ const create = defaults => {
 		return defaults.handler(url, options, next);
 	};
 
+	defaults.options.hooks = {
+		beforeRequest: [],
+		...(defaults.options.hooks || {})
+	};
+	got.hooks = defaults.options.hooks;
+
 	for (const method of defaults.methods) {
 		got[method] = (url, options) => got(url, {...options, method});
 		got.stream[method] = (url, options) => got.stream(url, {...options, method});
 	}
 
-	Object.assign(got, errors);
-	Object.defineProperty(got, 'defaults', {
-		value: deepFreeze(defaults),
-		writable: false,
-		enumerable: true,
-		configurable: true
-	});
+	defineConstProperty(got, {...errors, create}, {deepFreeze: true});
```

Comment:
I think this is overkill. If it's not overkill, I guess I'd say just call `defineConstProperty(got, got)` b/c it looks like `stream` and `extend` were arbitrarily omitted.

Reply:
Stream should be in. Extend no, because if someone doesn't like that function he can overwrite it :) What do you think?

Reply:
I would just get rid of it. When an object has a custom interface, those customizations should be documented, so that users have a way to learn about them. Documenting this stuff adds clutter to the documentation, adding friction for users.